src/main/java/de/dennisguse/opentracks/services/TrackRecordingManager.java

```diff
@@ -154,16 +161,28 @@ public Marker.Id insertMarker(String name, String category, String description,
         return new Marker.Id(ContentUris.parseId(uri));
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+    public void onIdle() {
+        Log.d(TAG, "Becoming idle");
+        onNewTrackPoint(trackPointCreator.createIdle());
+    }
+
     /**
      * @return TrackPoint was stored?
      */
-    boolean onNewTrackPoint(@NonNull TrackPoint trackPoint) {
+    synchronized boolean onNewTrackPoint(@NonNull TrackPoint trackPoint) {
         if (trackPoint.hasSpeed()) {
             lastTrackPointUIWithSpeed = trackPoint;
         }
         if (trackPoint.hasAltitude()) {
             lastTrackPointUIWithAltitude = trackPoint;
         }
+
+        if (trackPoint.getType() == TrackPoint.Type.IDLE) {
+            insertTrackPoint(trackPoint, true);
+            handler.removeCallbacks(this::onIdle);
```

Comment:
Interesting concept with the handler and the `postDelay()`.
Should `this::onIdle` be `ON_IDLE`?

Reply:
I wasn't sure how the handler keeps track of the callbacks, so I opted for a final variable to be on the safe side.
Will push -f it now.

Reply:
... done